src/App.tsx

```diff
@@ -1,25 +1,40 @@
 import React from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = React.useState<number>(5);
+  const [currentPage, setCurrentPage] = React.useState<number>(1);
+  const itemsListSize = items.length;
+
+  function onPerPageChange(e: React.ChangeEvent<HTMLSelectElement>): void {
+    setPerPage(+e.target.value);
+    setCurrentPage(1);
+  }
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {`${currentPage * perPage - perPage + 1}`} -{' '}
+        {`${currentPage < Math.ceil(itemsListSize / perPage) ? currentPage * perPage : itemsListSize}`}{' '}
+        of {`${itemsListSize}`})
```

Comment:
The current page calculation is a bit hard to read due to the inline arithmetic and ternary operator. It may be clearer to calculate these values before the return statement and store them in descriptive variables.

Reply:
done